### ISSUE TITLE
`ConnectionOpenTry`: `ValidateSelfClient` validation missing

### ISSUE THREAD

**Issue by plafer** [cosmos/ibc-rs#158]
## Summary of Bug
We are currently missing the entire `ValidateSelfClient` checks (see [ibc-go impl](https://github.com/cosmos/ibc-go/blob/a1843f8e240bf1c1de5df234b7d65e3cf6777393/modules/core/03-connection/keeper/handshake.go#L86) as reference).

____

## For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by plafer** [comment]
Blocked on #159.

**Comment by plafer** [comment]
duplicate of #162
